TestFramework.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author Michael Ray
+ * @since 3/22/2017
+ * @version 1.0.0
+ * 
+ */
+#include "TestFramework.hpp"
+
+/*/////////////////////////////////////////////////////////////////////////////
+//								Initialization								 //
+/////////////////////////////////////////////////////////////////////////////*/
+
+/**
+  * Constructor for the test framework. Add all of the PinMaps to the pinout, as well as CI test shield pins
+**/
+TestFramework::TestFramework() {
+	map_pins(PinMap_ADC, AnalogInput);
+	map_pins(PinMap_DAC, AnalogOutput);
```

Comment:
You should add some capability test like
#if !DEVICE_ANALOGOUT
for each map_pins call ?